messaging-client-databus/src/main/java/com/inmobi/databus/readers/CollectorStreamReader.java

```diff
@@ -144,6 +144,36 @@ protected CollectorFile getStreamFile(FileStatus file) {
     };
   }
 
+  public Long getPendingSize(Path readTill) throws IOException {
+    Long pendingSize = 0L;
+    Date readDoneTillDate = getDateFromCollectorFile(readTill.getName());
+    if (fsIsPathExists(streamDir)) {
+      FileStatus[] fileStatuses = fsListFileStatus(streamDir, null);
+      if (fileStatuses == null || fileStatuses.length == 0) {
+        LOG.info("No files in directory:" + streamDir);
+        return 0L;
+      }
+      for (FileStatus file : fileStatuses) {
+        Date currentTimeStamp = getDateFromCollectorFile(
+                file.getPath().getName());
+        if (stopTime == null) {
+          if (currentTimeStamp.before(readDoneTillDate)) {
+            continue;
+          }
+          pendingSize += file.getLen();
```

Comment:
you can move this outside if & else block.   you can remove *pendingSize += file.getLen();* from else block as well.